src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -9804,6 +9808,75 @@ HRESULT ProfToEEInterfaceImpl::ApplyMetaData(
     return hr;
 }
 
+// Add path to list of assemblies used by the runtime to probe assembly references
+// pAssemblyPath should be absolute path of assembly
+// Assembly name should match filename.
+HRESULT ProfToEEInterfaceImpl::AddAssemblyPath(AppDomainID appDomainId,
+    const WCHAR *pAssemblyPath)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
```

Comment:
BotR mention, that MODE_COOPERATIVE may be used if ObjectIDs are used. Not sure, what ObjectID means there. Here we use AppDomainID .